hudi-utilities/src/main/java/org/apache/hudi/utilities/sources/helpers/IncrSourceHelper.java

```diff
@@ -219,7 +218,7 @@ public static Pair<CloudObjectIncrCheckpoint, Dataset<Row>> filterAndGenerateChe
     }
     LOG.info("Processed batch size: " + row.get(row.fieldIndex(CUMULATIVE_COLUMN_NAME)) + " bytes");
     sourceData.unpersist();
-    return Pair.of(new CloudObjectIncrCheckpoint(row.getString(0), row.getString(1)), collectedRows);
+    return Pair.of(new CloudObjectIncrCheckpoint(row.getString(0), row.getString(1)), Option.of(collectedRows));
```

Comment:
Minor: Would suggest using row.fieldIndex just like what you have for `CUMULATIVE_COLUMN_NAME1`

Reply:
Also, please open a ticket for optimizations that we were discussing offline (Use of `applyOrdering` in line 182 and `commit_key` temporary column)

Reply:
https://issues.apache.org/jira/browse/HUDI-6687